demos/enclave/levi_model/compile_levi_model.py

```diff
@@ -0,0 +1,82 @@
+import os
+import pandas as pd
+from collections import defaultdict
+
+INPUT_PATH = os.getenv('DATA_INPUT_DIR')
+RESULTS_OUTPUT_PATH = os.getenv('RESULT_OUTPUT_DIR')
+
+quiz_questions = pd.read_csv(
+    f'{INPUT_PATH}/quiz_questions.csv'
+)
+quiz_question_contents = pd.read_csv(
+    f'{INPUT_PATH}/quiz_question_contents.csv'
+)
+quiz_multichoice_answers = pd.read_csv(
+    f'{INPUT_PATH}/quiz_multichoice_answers.csv'
+)
+
+merged_data = pd.merge(
+    quiz_questions, quiz_question_contents,
+    left_on="question_id", right_on='id'
+)
+merged_data.pop('id')
+merged_data.rename(columns={'text': 'question_text'}, inplace=True)
+
+answer_data = pd.merge(
+    quiz_multichoice_answers, merged_data,
+    on="question_id"
+)
+data_by_question = defaultdict(list)
+for index, row in answer_data.iterrows():
+    data_by_question[row['question_id']].append(row)
```

Comment:
This implicitly assumes that a question is never referenced multiple times. When that assumption is violated, duplicated questions end up being omitted in the final result. Unfortunately, due to the existing content :bug: we actually do hit that case in practice :grimacing: . Maybe using a tuple like `(quiz_name, question_number, question_id)` as the key would work instead :man_shrugging: (since, in theory, the same quiz could even reference the same question multiple times).